src/EditorFeatures/Core/ReferenceHighlighting/ReferenceHighlightingViewTaggerProvider.cs

```diff
@@ -171,11 +173,20 @@ protected override Task ProduceTagsAsync(TaggerContext<NavigableHighlightTag> co
                 return;
             }
 
-            foreach (var span in documentHighlights.HighlightSpans)
+            try
             {
-                var tag = GetTag(span);
-                context.AddTag(new TagSpan<NavigableHighlightTag>(
-                    textSnapshot.GetSpan(Span.FromBounds(span.TextSpan.Start, span.TextSpan.End)), tag));
+                foreach (var span in documentHighlights.HighlightSpans)
+                {
+                    var tag = GetTag(span);
+                    context.AddTag(new TagSpan<NavigableHighlightTag>(
+                        textSnapshot.GetSpan(Span.FromBounds(span.TextSpan.Start, span.TextSpan.End)), tag));
+                }
+            }
+            catch (Exception e) when (FatalError.ReportWithoutCrashUnlessCanceled(e))
+            {
+                // report NFW and continue.
+                // also, rather than return partial results, return nothing
+                context.ClearTags();
```

Comment:
this is super terrifying.  Do we have dumps that show what sort of document we have, and what the spans were in them?